### PR TITLE
🎨 Limit UI iteration (styles & functionality)

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Adds column 'Market price'
- Adds column 'Executes at'
- Adds column 'Filled'
- Small order warning banner + column indicator (+ tooltip)
- Tooltip hover on 'Executes at' price for open orders, indicating how close to executing the order is.
- Dot indicator for 'how close to executing' the order is.
- Explainer banner for Dot indicator functionality + column header tooltip
- Misc. style improvements on the order table layout.
- Adding back a close-able info banner below the limit order widget container.

Screenshot 2023-03-07 at 18 07 53

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @fairlighteth , just to clarify: should I focus here on styles only or test functionality as well?

**Review by elena-zh — CHANGES_REQUESTED** [review]
Anyways, just focusing on UI issues: 
1. I don't see 'Filled' column on the mock up (will be replaced with Execution time?)? As far as I can understand, amount it the column are not correct and require rework. Am I right? 
image

2. Also, small amount banner requires logic  implementation as not it is permanently displayed
3. Where Learn more links should navigate? 
image
image
4. And it would be great to add arrows for the Learn more links (to correspond to other links on the UI)
arrow
5. Warning icons are not in 1 line 
not in line
6. When hover a mouse on an amount in the dark mode, a newly implemented tooltip is almost invisible. Focus goes to the full amount on hover. Van we highlight the new tooltip somehow? 
diff to see
7. Scroll on the page appears when Open orders tab is opened, screen width is less than 1600 px (it is more than 14" screen). Can we adjust the table somehow to avoid this scrollbar displaying in a desktop view?
scroll
8. Banner messages are shifted on a reduced screen
reduced screen
9. In order to read the banner's message in a mobile view a view needs to scroll to the right. Can we show the ranges below the banner name in a mobile view? 
image
10. It is not related to the current PR, but I noticed that 10th page is displayed out of the screen area in a mobile view
10 is out

Thanks!

**Comment by fairlighteth** [comment]
> Hey @fairlighteth , just to clarify: should I focus here on styles only or test functionality as well?

Both please.

**Comment by elena-zh** [comment]
> > Hey @fairlighteth , just to clarify: should I focus here on styles only or test functionality as well?
> 
> Both please.

As for functionality: 
1. 'Small orders' banner is displayed all the time (when connected/disconnected/no token pair is selected/amount is not selected/price is not selected/amount is NOT small).  I think, we should show the banner when an amount is really small
2. Filled column on the Orders history tab shows not valid amounts
image
3. Tooltips are hardcoded 
image
image
4. I can't see blue/green indicators (I hope they will be fixed in another PR).

File: src/cow-react/modules/limitOrders/containers/LimitOrdersWidget/index.tsx [hunk 1]
```diff
@@ -292,6 +295,14 @@ const LimitOrders = React.memo((props: LimitOrdersProps) => {
                 </styledEl.FooterBox>
               )}
 
+              {/* TODO: Move this component inside LimitOrdersWarnings */}
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/containers/LimitOrdersWidget/index.tsx:298]: Just remainder about TODO, fix it when you wish

File: src/cow-react/modules/limitOrders/pure/InfoBanner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  // Explainer banner for orders
+  const LOCAL_STORAGE_KEY = 'limitOrders_showInfoBanner'
+  const [showBanner, setShowBanner] = useState(() => {
+    const item = localStorage.getItem(LOCAL_STORAGE_KEY)
+    return item !== null ? item === 'true' : true
+  })
+
+  const closeBanner = (): void => {
+    setShowBanner(false)
+    localStorage.setItem(LOCAL_STORAGE_KEY, 'false')
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/pure/InfoBanner/index.tsx:17]: Do we need this? Bellow we have useEffect that does the same

File: src/cow-react/modules/limitOrders/pure/Orders/OrderRow/index.tsx [hunk 3]
```diff
@@ -9,17 +9,24 @@ import { getSellAmountWithFee } from '@cow/modules/limitOrders/utils/getSellAmou
...
+import { TokenAmount, TokenAmountProps } from '@cow/common/pure/TokenAmount'
 import CurrencyLogo from 'components/CurrencyLogo'
 import useTimeAgo from 'hooks/useTimeAgo'
 import { ParsedOrder } from '@cow/modules/limitOrders/containers/OrdersWidget/hooks/useLimitOrdersList'
 import { OrderStatusBox } from '@cow/modules/limitOrders/pure/OrderStatusBox'
 import * as styledEl from './styled'
 import { getEtherscanLink } from 'utils'
 import { PendingOrderPrices } from '@cow/modules/orders/state/pendingOrdersPricesAtom'
-import Loader from 'components/Loader'
+import Loader from '@src/components/Loader'
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/pure/Orders/OrderRow/index.tsx:20]:
```suggestion
import Loader from 'components/Loader'
```

**Review by shoom3301 — APPROVED** [review]
Soft approve

File: src/cow-react/modules/limitOrders/pure/InfoBanner/index.tsx (hunk 2 above)
Comment (fairlighteth) [reply src/cow-react/modules/limitOrders/pure/InfoBanner/index.tsx:17]: Feel free to re-factor as I'm unsure what is needed or not.

**Comment by elena-zh** [comment]
Hey @fairlighteth , seems like the build is failed as I don't see changes in the PR

**Comment by elena-zh** [comment]
Hey @fairlighteth , great job!
Some nitpicks from my side: 
1. Seems like dots have different size on the screen 1536x864
diff size
2. I still see warning icons shifted from the value on reduced screens
reduced screen
3. Scroll under the pagination in Win 11 does not look good + pagination is left-aligned now
scrollbar
4. Sometimes it is a bit difficult to tap on the warning icon in a mobile view (need to do it twice). But it is a nitpick. 


Thanks!

**Comment by alfetopito** [comment]
@fairlighteth let's use the base branch `feature/limit-ui-update` for now on https://github.com/cowprotocol/cowswap/pull/2139

Contains everything in this PR plus latest develop.
Already addressed all the merge conflicts (there weren't many)

**Comment by fairlighteth** [comment]
@elena-zh Thank you again for your review:

1. Fixed.
2. Fixed: I now just right align them by default.
3. This is intentional and by design. On actual mobile devices you will not see the scrollbar (you are testing a mobile viewport on a non-mobile environment)
4. Made it bigger now, which should make it easier to tap on.
5. Fixed.

**Comment by alfetopito** [comment]
As agreed, merging to parent to consolidate.
Anything still pending will be addressed as follow up PR

**Pull request merged**